merlin/models/data/ecommerce/aliccp/dataset.py

```diff
@@ -0,0 +1,431 @@
+):
+    """
+    Convert Ali-CPP data to parquet files.
+
+    To download the raw the Ali-CCP training and test datasets visit
+    [tianchi.aliyun.com](https://tianchi.aliyun.com/dataset/dataDetail?dataId=408#1).
+
+    Parameters
+    ----------
+    data_dir: Union[str, Path]
+        Directory to load the raw data from.
+    convert_train: bool
+        Whether to convert the training data.
+    convert_test: bool
+        Whether to convert the test data.
+    file_size: int
+        Number of rows to write to each parquet file.
+    max_num_rows: int, optional
+        Maximum number of rows to read from the raw data.
+    pickle_common_features: bool
+        Whether to pickle the common features.
+        When enabled it will make the conversion faster if it would be run again.
+    output_dir: Union[str, Path], optional
+        Directory to write the converted data to.
+        If not specified the data will be written to the same directory as the raw data.
+
+    Returns
+    -------
+    data_dir
+    """
+
+    if convert_train:
+        _convert_data(
+            str(data_dir),
+            file_size,
+            is_train=True,
+            max_num_rows=max_num_rows,
+            pickle_common_features=pickle_common_features,
+            output_dir=output_dir,
+        )
+    if convert_test:
+        _convert_data(
+            str(data_dir),
+            file_size,
+            is_train=False,
+            max_num_rows=max_num_rows,
+            pickle_common_features=pickle_common_features,
+            output_dir=output_dir,
+        )
+
+    return data_dir
+
+
+def default_aliccp_transformation(add_target_encoding=True, **kwargs):
+    import nvtabular as nvt
+    from nvtabular import ops as nvt_ops
+
+    user_id = ["user_id"] >> nvt_ops.Categorify() >> nvt_ops.TagAsUserID()
+    item_id = ["item_id"] >> nvt_ops.Categorify() >> nvt_ops.TagAsItemID()
+
+    item_features = (
+        ["item_category", "item_shop", "item_brand"]
+        >> nvt_ops.Categorify(dtype="int32")
+        >> nvt_ops.TagAsItemFeatures()
+    )
+
+    user_features = (
+        [
+            "user_shops",
+            "user_profile",
+            "user_group",
+            "user_gender",
+            "user_age",
+            "user_consumption_2",
+            "user_is_occupied",
+            "user_geography",
+            "user_intentions",
+            "user_brands",
+            "user_categories",
+        ]
+        >> nvt_ops.Categorify(dtype="int32")
+        >> nvt_ops.TagAsUserFeatures()
+    )
+
+    targets = ["click", "conversion"] >> nvt_ops.AddMetadata(
+        tags=[str(Tags.BINARY_CLASSIFICATION), "target"]
+    )
+    outputs = user_id + item_id + item_features + user_features + targets
+
+    if add_target_encoding:
+        continuous = (
+            ["user_id", "item_id"]
+            >> nvt_ops.TargetEncoding(["click"], kfold=1, p_smooth=20)
+            >> nvt_ops.Normalize()
+        )
+        outputs += continuous
```

Comment:
Maybe we should tag the resulting columns as `user` or `item` features. Thinking of that, because we rely on these tags to select the features for the Two-Tower [model](https://github.com/NVIDIA-Merlin/models/blob/72ca5fb875e0513a4f5c0241a42c0d10ce1b16aa/merlin/models/tf/blocks/retrieval/two_tower.py#L101).